astropy/time/core.py

```diff
@@ -803,12 +803,15 @@ def _apply(self, method=None, *args, **kwargs):
             # Apply the method to any value arrays (though skip if there is only
             # a single element and the method would return a view, since in
             # that case nothing would change).
-            if method is not None and val is not None:
-                if method == 'copy' or method == 'flatten' and val.size == 1:
+            if method is not None and val is not None and hasattr(val, method):
```

Comment:
Yes, makes sense. Though I guess it is even better to just prefetch the method with `val_method = getattr(val, method, None)` and then test and use that. Will implement that.